# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1475734102 18000
#      Thu Oct 06 01:08:22 2016 -0500
# Branch stable
# Node ID 883b72cdf0b2563cae59c973e46d729df1e69b8f
# Parent  dd80fb1ef949cbd5c5846f0c51c4f12c03908971
Use old syntax to fix GPVAL_TERM conditionals under gnuplot 4.4.0 (bug #49223).

* __gnuplot_draw_axes__.m: Place all 'unset obj 1' gnuplot commands after a
  GPVAL_TERM conditional test to avoid extraneous gnuplot warning messages.

* __gnuplot_draw_figure__.m: Drop newer conditional syntax "{ }" in favor of
  old gnuplot syntax.  Remove an extraneous 'unset' command that would simply
  cause a gnuplot warning message.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1654,17 +1654,17 @@ function __gnuplot_draw_axes__ (h, plot_
     for i = 2:data_idx
       if (have_3d_patch (i))
         fprintf (plot_stream, ", \"-\" %s %s %s \\\n",
                  usingclause{i}, titlespec{i}, withclause{i});
       elseif (is_image_data (i))
         if (! is_image_data (i-1))
           fputs (plot_stream, "; ");
           if (bg_is_set)
-            fputs (plot_stream, "unset obj 1; \\\n");
+            fputs (plot_stream, "if (GPVAL_TERM eq \"qt\") unset obj 1;\n");
             bg_is_set = false;
           endif
           if (fg_is_set)
             fputs (plot_stream, "unset obj 2; \\\n");
             fg_is_set = false;
           endif
           if (numel (is_image_data) > i && is_image_data(i+1))
             ## Remove terminating semicolon
@@ -1677,17 +1677,17 @@ function __gnuplot_draw_axes__ (h, plot_
                    usingclause{i}, titlespec{i}, withclause{i});
         else
           ## For consecutive images continue with the same plot command
           fprintf (plot_stream, "%s \"-\" %s %s %s \\\n", ",",
                    usingclause{i}, titlespec{i}, withclause{i});
         endif
       elseif (is_image_data (i-1))
         if (bg_is_set)
-          fputs (plot_stream, "unset obj 1; \\\n");
+          fputs (plot_stream, "if (GPVAL_TERM eq \"qt\") unset obj 1;\n");
           bg_is_set = false;
         endif
         if (fg_is_set)
           fputs (plot_stream, "unset obj 2; \\\n");
           fg_is_set = false;
         endif
         fprintf (plot_stream,"%s \"-\" binary format='%%float64' %s %s %s \\\n",
                  plot_cmd, usingclause{i}, titlespec{i}, withclause{i});
@@ -1725,17 +1725,17 @@ function __gnuplot_draw_axes__ (h, plot_
   endif
 
   ## Needed to allow mouse rotation if gnuplot was put in map view.
   if (view_map && rot_x == 0 && rot_z == 0)
     fputs (plot_stream, "set view 0,0;\n");
   endif
 
   if (bg_is_set)
-    fputs (plot_stream, "unset obj 1;\n");
+    fputs (plot_stream, "if (GPVAL_TERM eq \"qt\") unset obj 1;\n");
     bg_is_set = false;
   endif
 
   fflush (plot_stream);
 
 endfunction
 
 function x = flip (x)
diff --git a/scripts/plot/util/private/__gnuplot_draw_figure__.m b/scripts/plot/util/private/__gnuplot_draw_figure__.m
--- a/scripts/plot/util/private/__gnuplot_draw_figure__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_figure__.m
@@ -33,21 +33,20 @@ function __gnuplot_draw_figure__ (h, plo
 
     if (nkids > 0)
       fputs (plot_stream, "\nreset;\n");
       fputs (plot_stream, "set autoscale keepfix;\n");
       fputs (plot_stream, "set origin 0, 0\n");
       fputs (plot_stream, "set size 1, 1\n");
       bg = get (h, "color");
       if (isnumeric (bg))
-        fprintf (plot_stream, "if (GPVAL_TERM eq \"qt\") {set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder}\n", round (255 * bg));
+        fprintf (plot_stream, "if (GPVAL_TERM eq \"qt\") set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder;\n", round (255 * bg));
         bg_is_set = true;
       else
         bg_is_set = false;
-        fprintf (plot_stream, "unset obj 1");
       endif
       fg_was_set = false;
 
       for i = nkids:-1:1
         type = get (kids(i), "type");
         switch (type)
           case "axes"
             if (strcmpi (get (kids (i), "tag"), "legend"))
